bin/rule-generator.js

```diff
@@ -0,0 +1,448 @@
+		validate: input => {
+			const checkName = input.toLowerCase();
+			// 1) check if valid rule name
+			if (
+				!(
+					checkName &&
+					checkName.length > 0 &&
+					checkName.search(/^[a-zA-Z-]+$/) !== -1
+				)
+			) {
+				throw new Error(
+					'CHECK name should not be empty and can contain alphabets and dashes.'
+				);
+			}
+			// 2) ensure no rule filename overlaps
+			if (axeChecksFileNames.includes(checkName)) {
+				throw new Error('CHECK name conflicts with an existing filename.');
+			}
+			// 3) ensure no rule id overlaps
+			if (axeChecksIds.includes(checkName)) {
+				throw new Error('CHECK ID already exists.');
+			}
+			return true;
+		}
+	},
+	GET_CHECK_CATEGORY: {
+		/**
+		 * Choose category for the CHECK? :
+		 * 	(Eg: aria, color, forms, label, language, media...)
+		 */
+		name: 'GET_CHECK_CATEGORY',
+		type: 'list',
+		message: 'Choose category for the CHECK?: ',
+		choices: [
+			'aria',
+			'color',
+			'forms',
+			'keyboard',
+			'label',
+			'language',
+			'lists',
+			'media',
+			'mobile',
+			'navigation',
+			'parsing',
+			'shared',
+			'tables',
+			'visibility'
+		]
+	},
+	GET_CHECK_TYPE: {
+		/**
+		 * Choose type of CHECK? :
+		 * 	(Eg: all, any, none)
+		 */
+		name: 'GET_CHECK_TYPE',
+		type: 'list',
+		message: 'Choose type for the CHECK?: ',
+		choices: ['all', 'any', 'none']
+	},
+	GET_IS_UNIT_TEST_ASSETS: {
+		/**
+		 * Create UNIT test files? (Yes/ No)
+		 */
+		name: 'GET_IS_UNIT_TEST_ASSETS',
+		type: 'confirm',
+		message: 'Would you like to create UNIT test files?'
+	},
+	GET_IS_INTEGRATION_TEST_ASSETS: {
+		/**
+		 * Create INTEGRATION test files? (Yes/ No)
+		 */
+		name: 'GET_IS_INTEGRATION_TEST_ASSETS',
+		type: 'confirm',
+		message: 'Would you like to create INTEGRATION test files?'
+	}
+};
+
+const questionAndGetFilesToCreate = async () => {
+	const getCheck = async (checks = []) => {
+		const checkDetails = await inquirer.prompt([
+			QUESTIONS.GET_CHECK_NAME,
+			QUESTIONS.GET_CHECK_CATEGORY,
+			QUESTIONS.GET_CHECK_TYPE,
+			QUESTIONS.GET_IS_ANOTHER_CHECK
+		]);
+		checks.push(checkDetails);
+		if (checkDetails.GET_IS_ANOTHER_CHECK) {
+			await getCheck(checks);
+		}
+		return checks;
+	};
+
+	// get rule meta
+	const { GET_RULE_NAME, GET_IS_RULE_MATCHES } = await inquirer.prompt([
+		QUESTIONS.GET_RULE_NAME,
+		QUESTIONS.GET_IS_RULE_MATCHES
+	]);
+	// get checks meta
+	let ruleChecks = [];
+	const { GET_IS_CHECK } = await inquirer.prompt([QUESTIONS.GET_IS_CHECK]);
+	if (GET_IS_CHECK) {
+		ruleChecks = await getCheck();
+	}
+	// get if test files should be created
+	const {
+		GET_IS_UNIT_TEST_ASSETS,
+		GET_IS_INTEGRATION_TEST_ASSETS
+	} = await inquirer.prompt([
+		QUESTIONS.GET_IS_UNIT_TEST_ASSETS,
+		QUESTIONS.GET_IS_INTEGRATION_TEST_ASSETS
+	]);
+
+	let filesToGenerate = [];
+
+	const ruleName = `${GET_RULE_NAME.toLowerCase()}`;
+
+	const ruleJson = {
+		name: `${ruleName}.json`,
+		content: JSON.stringify(
+			{
+				id: `${ruleName}`,
+				selector: '',
```

Comment:
Should there be a prompt for the `selector` so new users don't forget to change it? I'm thinking of the Hackathon scenario. Same for tags and metadata. Maybe we could add a flag to skip certain fields in case you don't want to fill them all out, sort-of like the behavior of `npm init -y`.

Reply:
@marcysutton 
Can we add these as new requirements please.
All of the above are possible.

Just create an issue and assign to me. I will prioritise when I have downtime.